src/rules/no-extraneous-dependencies.js

```diff
@@ -121,11 +124,16 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
   const isInDevDeps = deps.devDependencies[packageName] !== undefined
   const isInOptDeps = deps.optionalDependencies[packageName] !== undefined
   const isInPeerDeps = deps.peerDependencies[packageName] !== undefined
+  // BundledDeps should be in the form of an array, but object notation is also supported by `npm`
+  const isInBundledDeps =
+    deps.bundledDependencies.indexOf(packageName) !== -1 ||
```

Comment:
objects don't have an `indexOf` method, so this will crash.

Reply:
Hmm, I actually had tests for this scenario. Funnily, the object `deps.bundledDependencies` was something in between an object and an array, which can be reconstructed like this:

```javascript
const arrayObject = Object.assign([], {a: 3});
console.log(Array.isArray(arrayObject)) // true
console.log(typeof arrayObject) // "object"
console.log(arrayObject.c) // 3
console.log(arrayObject) // [a: 3]
console.log(arrayObject[0]) // undefined 
```

I'm not sure if this is a bug in JavaScript by itself or just a weird edge-case, I've never saw something like this.

Anyways, I clean the `bundledDependencies` now so it is always an array and go with `indexOf`.

Reply:
That's not a bug; all arrays are objects.